test/distributed/_tensor/test_random_ops.py

```diff
@@ -126,6 +175,87 @@ def test_deterministic_dropout_1d(self):
                         local_tensor[self_slice, :], local_tensor[other_slice, :]
                     )
 
+    @with_comms
+    @skip_unless_torch_gpu
+    def test_deterministic_uniform_nd(self):
+        mesh = torch.arange(self.world_size).reshape(2, 2, -1)
```

Comment:
iirc the default world size of a TestCase is 4, if you want to test world size of 8, better have a separate TestCase that overrides the world_size, otherwise if the world_size is 4, the mesh of this test would become (2, 2, 1), which means the sharding on the last one invalid.

Reply:
btw, how is the current CI status now? I recall that we had a time that CI didn't run with 8 GPUs and all tests requiring 8 ranks only ran with 8 processes on CPU and skipped the GPU testing.

Reply:
No CI runs 8 GPUs (unless there's a periodic CI that have 8 GPUs), but for other tests the CI would run with 8 CPU processes. Given this also skip if there's no GPU, I think it would always be skipped unfortunately. 

We would need to add this test to the peridic CI so that there's some CI signal, you can add it following this PR https://github.com/pytorch/pytorch/pull/96711